elastic/kibana
# [AO] Fix showing temporary data view in the list of data views for the new threshold rule

Closes #159774, closes #159778, closes #159779, closes #159776

## Summary

This PR fixes the data view list in the new metric threshold. Also, it adds an API integration test to check the reference in the rule saved object.

Also, this PR improves the error handling and loading of the data view, similar to what we have for elasticsearch rule data view.
|Threshold rule error|Elasticsearch query error|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/ae5ff9bb-8bcc-46d2-8b84-42f7128f18e1)|![image](https://github.com/elastic/kibana/assets/12370520/1f65ff6b-18f3-4a4d-941b-cafe35c4d145)|

## How to test
- Make sure `xpack.observability.unsafe.thresholdRule.enabled` is set to true in kibana yml config

Error handling
- Throw an error [here](https://github.com/elastic/kibana/pull/161462/files#diff-4f65f6debaf6457d4b0400a27c1ea57ba52bfe4426ee40460d43a857c5bd165eL98) and make sure the message is shown correctly in the rule.

Temporary data view
- Create a temporary data view
- Check the list of data views and make sure the new temporary one is added to the list

|Adding a new temporary data view|Temporary data view in the list|
|---|---|
|![image](https://github.com/elastic/kibana/assets/12370520/5f0866b6-34ad-4072-9a84-48e0760072d6)|![image](https://github.com/elastic/kibana/assets/12370520/9109308e-88d4-49f7-98b1-19f635804c48)|
